common/src/main/java/org/dash/wallet/common/ui/components/MenuItem.kt

```diff
@@ -0,0 +1,195 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package org.dash.wallet.common.ui.components
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.*
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberUpdatedState
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import org.dash.wallet.common.R
+
+//@Composable
+//fun MenuItemIcon(icon: Int?) {
+//    icon?.let {
+//        Image(
+//            painter = painterResource(id = it),
+//            contentDescription = null,
+//            modifier = Modifier.size(36.dp)
+//        )
+//    }
+//}
+//
+//@Composable
+//fun MenuItemTexts(title: String, subtitle: String?, details: String?) {
+//    Column(
+//        modifier = Modifier
+//            .weight(1f)
+//            .padding(start = 8.dp)
+//    ) {
+//        Text(
+//            text = title,
+//            fontWeight = FontWeight.Medium,
+//            fontSize = 14.sp,
+//            color = Color.Black,
+//            modifier = Modifier.weight(1f)
+//        )
+//
+//        subtitle?.let {
+//            Text(
+//                text = it,
+//                fontSize = 12.sp,
+//                color = Color.DarkGray,
+//                modifier = Modifier.padding(top = 2.dp)
+//            )
+//        }
+//
+//        details?.let {
+//            Text(
+//                text = it,
+//                fontSize = 12.sp,
+//                color = Color.DarkGray,
+//                modifier = Modifier.padding(top = 2.dp)
+//            )
+//        }
+//    }
+//}
+
+@Composable
+fun MenuItem(
+    title: String,
+    subtitle: String? = null,
+    details: String? = null,
+    icon: Int? = null, // Assuming you use resource IDs for icons
+    showInfo: Boolean = false,
+    showChevron: Boolean = false,
+    isToggled: (() -> Boolean)? = null, // Change to a lambda that returns a Boolean
+    onToggleChanged: ((Boolean) -> Unit)? = null,
```

Comment:
ChatGPT comments can be removed.

I think it came up with something weird here and below with `toggleState`. 
`isToggled` should not be lambda if we already have `onToggleChanged` for reporting state change. It should just be `isToggled: Boolean = false,`.

The reason the binding in iOS code is nullable is to indicate if we need to show the toggle in the first place. If we don't need the toggle, binding is null. If we do and it's not null, we bind it to the caller's state.

In JC, we can either use `MutableState` or split it in two like it's sort of done here, but indicate visibility with the `onToggleChanged` callback. If it's nullable, don't show the toggle. Like this:

 ```
onToggleChanged?.let {
    Switch(
        checked = isToggled,
        onCheckedChange = { newState ->
            onToggleChanged?.invoke(newState)
        },
        modifier = Modifier.size(60.dp)
    )
}
```

Reply:
thanks